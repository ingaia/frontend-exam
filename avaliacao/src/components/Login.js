import React, { Component } from 'react';
import logoDark from '../assets/images/logo_dark.png';
import Loading from './Loading';
import {withRouter} from 'react-router-dom';

import '../styles/Login.scss';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      logged: false,
      username: '',
      password: '',
      loader: true,
      error: false
    }
  }

  doLoading = () => {
    if(this.state.loader) {
      let overlay = document.getElementById("loading-overlay");
      setTimeout(() => overlay.classList.add("hide"), 4000);
    }
  }


  componentDidMount() {
    document.getElementsByTagName("BODY")[0].classList.add("login-page");
    document.getElementsByTagName("BODY")[0].classList.remove("intern-page");
    this.doLoading();
  }

  handleChangeUsername = (event) => {
    this.setState({ username: event.target.value })
  }

  handleChangePassword = (event) => {
    this.setState({ password: event.target.value })
  }

  doLogin = () => {
    if (this.state.username.length > 0  && this.state.password.length > 0) {
      this.setState({ logged: true });
      document.getElementsByTagName("BODY")[0].classList.add("intern-page");
      return this.props.history.push('/intern');
    } else {
        this.setState({ error: true });
    }
  }

  render() {
    return (
      <section className="login-container">
          <div className="login-overlay"></div>
          <div className="form-container">
            <img src={logoDark} alt="Logo The Witcher 3"/>
            <form className="login-form" onSubmit={e => e.preventDefault(e)}>
              <fieldset>
                <label>
                    {
                        this.state.username.length > 0 &&
                        <p>Email</p>
                    }
                    <input className="login-username"
                    placeholder="Email"
                    onChange={this.handleChangeUsername} 
                    value={this.state.username} />
                </label>

                <label>
                    {
                        this.state.password.length > 0 &&
                        <p>Password</p>
                    }
                    <input className="login-username"
                    placeholder="Password"
                    type="password"
                    onChange={this.handleChangePassword} 
                    value={this.state.password} />
                </label>

                <p className={this.state.error ? "error-message" : "error-message hidden"}>Preencha os dois campos com alguma informação</p>

                <button onClick={this.doLogin}>Login</button>
              </fieldset>
            </form>
          </div>
          <Loading />
      </section>
    );
  }
}

export default withRouter(Login);