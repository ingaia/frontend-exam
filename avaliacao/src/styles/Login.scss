@import "Variables.scss";

.login-container {
    position: relative;
    display: block;
    min-height: 400px;
    height: 100%;

    .form-container {
        position: absolute;
        display: block;
        width: 300px;
        height: 380px;
        top: 50%;
        left: 50%;
        margin-top: -190px;
        margin-left: -150px;

        img {
            width: 100%;
            max-width: 300px;
            margin-bottom: 50px;
        }

        .login-form {
            fieldset {
                border: 0;
                
                p {
                    font-size: 11px;
                    color: $primaryColor;
                }

                input {
                    font-size: 13px;
                    color: $white;
                    width: 300px;
                    padding-bottom: 5px;
                    margin-bottom: 15px;
                    border: 0;
                    background-color: transparent;
                    border-bottom: 1px solid $primaryColor;

                    &::placeholder {
                        font-size: 14px;
                        color: $primaryColor;
                    }

                    &:focus {
                        border-bottom: 2px solid $primaryColor;
                    }
                }

                p.error-message {
                    display: block;
                    width: 300px;
                    margin: 0 auto;
                    padding: 5px;
                    color: $white;
                    background-color: $secondaryColor;
                    border-radius: 3px;
                    text-align: center;

                    &.hidden {
                        display: none;
                    }
                }

                button {
                    display: block;
                    width: 100px;
                    font-size: 14px;
                    color: $primaryColor;
                    margin: 0 auto;
                    margin-top: 30px;
                    padding: 10px;
                    background-color: transparent;
                    border: 1px solid $primaryColor;

                    &:hover {
                        cursor: pointer;
                        border: 2px solid $primaryColor;
                        transition: all .2s ease;
                    }
                }
            }
        }
    }
}