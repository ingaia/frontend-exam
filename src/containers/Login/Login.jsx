import React, {useState} from 'react';
import LogoDark from '../../assets/images/logo_dark.png';
import TextField from "../../components/TextField/TextField";
import {ButtonLogin, LoginStyled} from './styled';
import Background from "../../components/Background/Background";
import Box from "../../components/Box/Box";

const Login = ({history}) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleEmailChange = e => {
    setEmail(e.target.value);
  };

  const handlePasswordChange = e => {
    setPassword(e.target.value);
  };

  const doLogin = () => {
    localStorage.setItem('authToken', 'FAKE_TOKEN');
    history.push(`/`);
  };

  return (
    <Background>
      <LoginStyled>
        <Box>
          <img src={LogoDark} alt={`Logo`}/>
          <TextField
            id={`email`}
            name={`email`}
            label={`email`}
            type={`text`}
            value={email}
            onChange={handleEmailChange}/>
          <TextField
            id={`password`}
            name={`password`}
            label={`password`}
            type={`password`}
            value={password}
            onChange={handlePasswordChange}/>
          <ButtonLogin
            type="submit"
            variant="contained"
            name={'login'}
            value={'login'}
            text={'login'}
            onClick={doLogin}
          />
        </Box>
      </LoginStyled>
    </Background>
  );
};

export default Login;