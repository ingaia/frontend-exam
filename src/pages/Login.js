import React from "react";
import {Button, Form} from "react-bootstrap";
import {withRouter} from "react-router-dom";

const Login = ({history}) => {
  const onClick = () => history.push('library')
  return (
    <div className={'login'}>
      <img src='./assets/logo_dark.png' alt='witcher logo'/>
      <div className='form-login'>
        <Form.Label>Email address</Form.Label>
        <Form.Control type="email" placeholder="Enter email"/>
        <Form.Label>Password</Form.Label>
        <Form.Control type="password" placeholder="Password" />
        <Button onClick={onClick}>Login</Button>
      </div>
    </div>)
}

export default withRouter(Login);
