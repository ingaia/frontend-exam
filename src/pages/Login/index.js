import React, { useState } from 'react';

import InputText from '../../components/InputText';
import Button from '../../components/Button';

import './style.less';
import logo_dark from '../../../assets/logo_dark.png';

export default function Login({ history }){
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    function handleSubmit(event){
        event.preventDefault();
        history.push('/trailers');
    }

    return (
        <div className="login">
            <div className="box-login">
                <img className="logo" src={logo_dark} alt=""/>
                <form onSubmit={handleSubmit} name="formLogin" className="form-login">
                    <InputText type="email" name="email" label="Email" 
                        value={email} onChange={event => setEmail(event.target.value)} required={true}/>
                    <InputText type="password" name="password" label="Password" 
                        value={password} onChange={event => setPassword(event.target.value)} required={true}/>
                    <Button type="submit" className="btn-hillary" label="Login" />
                </form>
            </div>
        </div>
    )
}