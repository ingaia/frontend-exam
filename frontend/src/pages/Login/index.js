import React, { useState } from "react";

import logo from '../../assets/logo_dark.png'
import './styles.css'

export default function Login({history}) {

    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    function handlerSubmit(event){
        event.preventDefault();

        history.push('/dashboard');
    }
    
    function validateForm() {
       return email.length > 0 && password.length > 0;
    }
  
    
    return (
        <>
        <div className="content" >
            <div className="logo" >
                <img src={logo} alt="The Witcher"/>
            </div>

            <div className="form__wrapper" >
                <form onSubmit={handlerSubmit}>
                    <ul>
                        <li>
                            <label htmlFor="email">Email</label>
                            <input
                                autoFocus
                                type="email"
                                value={email}
                                onChange={event => setEmail(event.target.value)}
                            />
                        
                        </li>
                        <li>
                            <label htmlFor="email">Password</label>
                            <input
                                value={password}
                                onChange={event => setPassword(event.target.value)}
                                type="password"
                            />
                        </li>
                        <li>
                            <div className="button__form">
                                <button
                                    className="btn__default"
                                    disabled={!validateForm()}
                                    type="submit"
                                >
                                    Login
                                </button>
                            </div>
                        </li>
                    </ul>
                </form>
            </div>
        </div>
        </>
    )
    
}
