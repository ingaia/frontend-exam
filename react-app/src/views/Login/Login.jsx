import React, { useState } from "react";

import classnames from "classnames";
import Logo from "components/Logo";
import TextInput from "components/TextInput";
import { useUser } from "UserContext";
import { useHistory } from "react-router-dom";
import styles from "./Login.module.scss";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [userData, setUserData] = useUser();

  const history = useHistory();
  return (
    <div className={styles.loginPanel}>
      <Logo className={styles.logo} />
      <TextInput
        id="emailInput"
        label="Email"
        className={styles.email}
        value={email}
        onChange={evt => setEmail(evt.target.value)}
        type="email"
      />
      <TextInput
        id="passwordInput"
        label="Password"
        className={styles.password}
        value={password}
        onChange={evt => setPassword(evt.target.value)}
        type="password"
      />
      <button
        className={classnames(styles.submitLogin)}
        onClick={() => {
          setUserData({ isAuthenticated: true });
          history.push("/ListVideos");
        }}
      >
        LOGIN
      </button>
    </div>
  );
};

export default Login;
