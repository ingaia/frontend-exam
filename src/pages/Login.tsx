import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { isEmpty, trim } from 'lodash';

import actions from '../store/ducks/auth/actions';
import { typedUseSelector } from '../store';

import Grid from '@material-ui/core/Grid';

import TextField from '../components/TextField';
import Button from '../components/Button';
import Layout from '../components/Layout';
import Loader from '../components/Loader';

import * as S from '../styles/Login';

const Login = () => {
  let history = useHistory();
  const dispatch = useDispatch();
  const state = typedUseSelector((state) => state.auth);
  const [values, setValues] = useState({ email: '', password: '' });

  useEffect(() => {
    if (state.isLogged) {
      history.push('/');
    }
  }, [state.isLogged]);

  const isValid = () => {
    return !isEmpty(trim(values.email)) && !isEmpty(trim(values.password))
  }
  const onSubmit = (event: any) => {
    event.preventDefault();
    if (isValid()) {
      dispatch(actions.loginRequest());
    }
  };

  const handleChange = (prop: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  if (state.loading) {
    return <Loader />;
  }

  return (
    <Layout>
      <S.ContainerWrapper disableGutters maxWidth={false}>
        <Grid container justify='center' alignItems='center' className='bg-opacity'>
          <Grid item xs={12} md={3}>
            <Grid container>
              <Grid item xs={12}>
                <S.Logo
                  src='/assets/logo_dark.png'
                  alt='logo'
                />
              </Grid>
              <Grid item xs={12}>
                <form noValidate autoComplete='off' onSubmit={onSubmit}>
                  <Grid container justify='center'>
                    <Grid item xs={12}>
                      <TextField
                        fullWidth
                        id='email'
                        type='email'
                        label='Email'
                        value={values.email}
                        onChange={handleChange('email')}
                      />
                    </Grid>
                    <Grid item xs={12}>
                      <TextField
                        fullWidth
                        id='password'
                        type='password'
                        label='Password'
                        value={values.password}
                        onChange={handleChange('password')}
                      />
                    </Grid>
                    <Grid item xs={12} md={4} xl={2}>
                      <Button fullWidth type='submit'>
                        Login
                      </Button>
                    </Grid>
                  </Grid>
                </form>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </S.ContainerWrapper>
    </Layout>
  );
};

export default Login;
