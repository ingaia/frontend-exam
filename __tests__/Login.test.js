import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import Login from '../src/pages/Login';
import history from '../__mocks__/HistoryMock'

describe('<Login/>', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Login history={history}/>);
    });

    it('should render', () => {
        expect(wrapper.hasClass('login')).toBe(true);
        expect(wrapper.find('.box-login').length).toBe(1);
        expect(wrapper.find('.logo').length).toBe(1);
        expect(wrapper.find('.form-login').length).toBe(1);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should call submit()', () => {
        const form = wrapper.find('form')
        jest.spyOn(history, 'push');
        
        form.simulate('submit', {
            preventDefault: () => {}
        });
        
        expect(history.push).toHaveBeenCalled();
    });
});