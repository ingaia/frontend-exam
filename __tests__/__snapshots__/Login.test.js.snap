// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render 1`] = `
<div
  className="login"
>
  <div
    className="box-login"
  >
    <img
      alt=""
      className="logo"
      src="test-file-stub"
    />
    <form
      className="form-login"
      name="formLogin"
      onSubmit={[Function]}
    >
      <InputText
        label="Email"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
      <InputText
        label="Password"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <Button
        className="btn-hillary"
        label="Login"
        type="submit"
      />
    </form>
  </div>
</div>
`;
